com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/resource/StrongReferencerCacheAdapter.java

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Avaloq Evolution AG and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Avaloq Evolution AG - initial API and implementation
+ *******************************************************************************/
+package com.avaloq.tools.ddk.xtext.resource;
+
+import com.avaloq.tools.ddk.caching.CacheConfiguration;
+
+
+/**
+ * A generic resource set cache adapter which uses strong references.
+ *
+ * @param <V>
+ *          the type of the stored values
+ */
+public class StrongReferencerCacheAdapter<V> extends CacheAdapter<V> {
+  @Override
+  protected CacheConfiguration getCacheConfiguration() {

```

Comment:
be aware that in general invoking the protected/public methods from constructors/initializers isn't a safe pattern, as these methods can be overridden and invoke while the object isn't completely initialized